### PR TITLE
Kotlin script 1.0.1

### PR DESCRIPTION
**Pull request by jakob-grabner**
TagCheck

# Add new kotlin script step

Allows to run kotlin script files.

### New Pull Request Checklist

*Please mark the points which you did / accept.*

- [x] __I will not move an already shared step version's tag to another commit__
- [x] I read the [Step Development Guideline](https://github.com/bitrise-io/bitrise/blob/master/_docs/step-development-guideline.md)
- [x] I have a test for my Step, which can be run with `bitrise run test` (in the step's repository)
- [x] I did run `bitrise run audit-this-step` (in the step's repository - note, if you don't have this workflow in your `bitrise.yml`, [you can copy it from the step template](https://github.com/bitrise-steplib/step-template/blob/master/bitrise.yml).)
- [x] I read and accept the [Abandoned Step policy](https://github.com/bitrise-io/bitrise-steplib#abandoned-step-policy)

**New Step**
Thank you for the new Step share! The CI check might will fail due to our extended validation engine. Nothing to worry about yet, we will get back to you shortly.

### CODE REVIEW

File: steps/kotlin-script/1.0.0/step.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+published_at: 2023-10-18T15:02:52.092092+02:00
+source:
+  git: https://github.com/jakob-grabner/bitrise-step-kotlin-script.git
+  commit: 08ea2e8f750f6546c1da89b6387956a59173357a
+type_tags:
+- utility
+toolkit:
+  bash:
+    entry_file: step.sh
+deps:
```
Comment (ofalvai) [inline steps/kotlin-script/1.0.0/step.yml:18]: Kotlin is installed on both macOS and Linux stacks and it's installed via other means, so you can probably remove this deps block and save some unnecessary install time

File: steps/kotlin-script/1.0.0/step.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+inputs:
+- kotlin_content: |
+    println("Hello World!")
+  opts:
+    description: The Kotlin script to run
+    is_expand: false
+    is_required: true
+    summary: The Kotlin script to run
+    title: The Kotlin script
+- opts:
```
Comment (ofalvai) [inline steps/kotlin-script/1.0.0/step.yml:32]: Please make the default value of this input `$BITRISE_SOURCE_DIR`, by convention most steps use this directory as the default workdir.

**Review by ofalvai — CHANGES_REQUESTED** [review]
Thank you @jakob-grabner for the contribution! 🚀 
Looks good overall, I just flagged a few minor things.

**Comment by ofalvai** [comment]
Oh and please only submit one step version per PR. It's okay if the first submitted version is not 1.0.0.

File: steps/kotlin-script/1.0.0/step.yml (hunk 2 above)
Comment (jakob-grabner) [reply steps/kotlin-script/1.0.0/step.yml]: I oriented this step on the ruby script step, there it was also empty. But changed it now to BITRISE_SOURCE_DIR

**Comment by jakob-grabner** [comment]
@ofalvai I finally had some time to work on your comments. Please have another look :)

**Comment by jakob-grabner** [comment]
@ofalvai any chance we can get this merged?

**Review by ofalvai — APPROVED** [review]
Looking good now @jakob-grabner, thank you for the contribution 🚀 💜

**Pull request merged**